### ISSUE TITLE
Define LinearBandit parameters as nn.Parameter

### ISSUE THREAD

**Issue by philippkolbe** [neural-bandits/calvera#44]
We could think about properly defining the parameters of the LinearBandit as torch parameters:
```
class LinearBandit(AbstractBandit):
    def __init__(self, n_features: int) -> None:
        super().__init__(n_features)

        self.precision_matrix: torch.Tensor = torch.nn.Parameter(
            torch.eye(n_features), requires_grad=False
        )
        self.b = torch.nn.Parameter(torch.zeros(n_features), requires_grad=False)
        self.theta = torch.nn.Parameter(torch.zeros(n_features), requires_grad=False)

```

This way they will show up as Non-trainable parameters:
<img width="260" alt="Image" src="https://github.com/user-attachments/assets/71c8634b-869d-49fc-a4be-73cda927d607" />

Otherwise, we would have only 0s there.
The non-trainable parameters come from the "requires_grad=False". I think this is expected because we do our own training step but its also kind of weird. Maybe it should actually be "requires_grad=True"?

**Comment by Rob2U** [comment]
This should already be done (as all parameters should be registered via `register_buffer` because they are not trained via gradients)

**Comment by philippkolbe** [comment]
<img width="424" alt="Image" src="https://github.com/user-attachments/assets/f3237812-0e67-45dd-80d4-fb0b1373c47d" />

**Comment by Rob2U** [comment]
So in this Issue you should check if `self.register_buffer` is correct or if we should use `nn.Parameter` or `self.register_buffer`
